mne/utils/config.py

```diff
@@ -238,9 +242,8 @@ def set_config(key, value, home_dir=None, set_env=True):
 
     Parameters
     ----------
-    key : str | None
```

Comment:
you change public API?

Reply:
yes.  Current API is bad because to *get* the valid keys, you call the *setting* function (`set_config(key=None, value=None)`).  So new API is that the *setting* function is only ever used to set keys, so `set_config()` is no longer allowed to take `key=None`.

But you raise a good point, this needs a deprecation cycle?

Reply:
Yes :(